From 69b85c477af2237cc4d3e2a9573047d3d528726b Mon Sep 17 00:00:00 2001
From: Min He <min.he@intel.com>
Date: Thu, 18 Jan 2018 15:15:53 +0800
Subject: [PATCH 0966/2313] drm/i915/gvt: fix the issue in "preliminary per
 ring scheduler"

This patch fixed an issue in "preliminary per ring scheduler", which
causes the guest GPU hang when submitting on engines other than the
render engine.

Change-Id: If18266325dc9624dc85e1e1557054f18c7e095c7
Signed-off-by: Min He <min.he@intel.com>
---
 drivers/gpu/drm/i915/gvt/sched_policy.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/gvt/sched_policy.c b/drivers/gpu/drm/i915/gvt/sched_policy.c
index 02e0915..4dbb068 100644
--- a/drivers/gpu/drm/i915/gvt/sched_policy.c
+++ b/drivers/gpu/drm/i915/gvt/sched_policy.c
@@ -83,8 +83,8 @@ static void gvt_balance_timeslice(struct gvt_sched_data *sched_data,
 {
 	struct vgpu_sched_data *vgpu_data;
 	struct list_head *pos;
-	static uint64_t stage_check;
-	int stage = stage_check++ % GVT_TS_BALANCE_STAGE_NUM;
+	static uint64_t stage_check[I915_NUM_ENGINES];
+	int stage = stage_check[ring_id]++ % GVT_TS_BALANCE_STAGE_NUM;
 
 	/* The timeslice accumulation reset at stage 0, which is
 	 * allocated again without adding previous debt.
-- 
2.7.4

